### PR TITLE
Open cloud audio

### PR DESCRIPTION
**Pull request by kryksyh**
Resolves: #9776, #9874, #9862

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

QA:

- [ ] Autobot test cases have been run

### CODE REVIEW

File: libraries/lib-cloud-audiocom/TaskExecutionService.cpp [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+namespace audacity::cloud::audiocom
+{
+
+namespace {
+
+std::string TaskStatusToString(sync::TaskStatus status)
+{
+   switch (status)
+   {
+      case sync::TaskStatus::Success:
```
Comment (embarc-gabriel) [inline libraries/lib-cloud-audiocom/TaskExecutionService.cpp:38]: Missing Pending?

File: libraries/lib-cloud-audiocom/TaskExecutionService.cpp [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+}
+
+TaskExecutionService::~TaskExecutionService()
+{
+   mStopRequested = true;
+
+   if (mPollingThread.joinable())
+   {
+      mPollingThread.detach();
+   }
```
Comment (embarc-gabriel) [inline libraries/lib-cloud-audiocom/TaskExecutionService.cpp:81]: Shouldn't it just call `Stop()` and then `detach` instead?
I mean, imagine this scenario:
PollingThread tests mStopRequested in the while loop, which is false, for example. It enters the code inside, and then the destructor is called, and it sets `mStopRequested` to `true`. But the code PollingThread is still being executed and will test the atomic variable just on the next iteration.
This can be a problem, in my opinion.

File: libraries/lib-cloud-audiocom/sync/CloudSyncDTO.cpp [hunk 3]
```diff
@@ -462,6 +462,146 @@ bool Deserialize(
...
+   else
+      taskAction = TaskAction::Unknown;
+
+   return true;
+}
+
+bool Deserialize(const rapidjson::Value& value, AppTask& task)
+{
+   if (!value.IsObject())
+      return {};
```
Comment (embarc-gabriel) [inline libraries/lib-cloud-audiocom/sync/CloudSyncDTO.cpp]: Nit.
I would prefer `false` instead of {}.
I know for bool type `{}` == false, but I think it is clear to use `false`

File: libraries/lib-cloud-audiocom/TaskExecutionService.cpp (hunk 2 above)
Comment (kryksyh) [reply libraries/lib-cloud-audiocom/TaskExecutionService.cpp:81]: Yes, you are right, the thing is that it is called on the app shutdown, and there is not much time to wait for the thread stop. I'll add a small timeout.

**Review by embarc-gabriel — APPROVED** [review]
